internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -210,6 +212,39 @@ func (c *CodePipeline) ListPipelineNamesByTags(tags map[string]string) ([]string
 	return pipelineNames, nil
 }
 
+// ListPipelineExecution returns the ExecutionID of the most recent execution of a pipeline.
+func (c *CodePipeline) ListPipelineExecution(pipelineName string) (*string, error) {
+	input := &cp.ListPipelineExecutionsInput{
+		MaxResults:   aws.Int64(1),
+		PipelineName: &pipelineName,
+	}
+	output, err := c.client.ListPipelineExecutions(input)
+	if err != nil {
+		return nil, fmt.Errorf("list pipeline execution for %s: %w", pipelineName, err)
+	}
+	return output.PipelineExecutionSummaries[0].PipelineExecutionId, nil
+}
+
+// RetryStageExecution tries to rebuild a failed pipeline.
+func (c *CodePipeline) RetryStageExecution(pipelineName string) error {
+	executionID, err := c.ListPipelineExecution(pipelineName)
+	if err != nil {
+		return fmt.Errorf("retrieve pipeline execution ID: %w", err)
+	}
+	input :=
+		&cp.RetryStageExecutionInput{
+			PipelineExecutionId: executionID,
+			PipelineName:        &pipelineName,
+			RetryMode:           aws.String("FAILED_ACTIONS"),
```

Comment:
nit: The `codepipeline` pkg provides the const `cp.StageRetryModeFailedActions` we can used that instead